### ISSUE TITLE
CourseRestore.php with overwrite option

### ISSUE THREAD

**Issue by csch1** [tmuras/moosh#151]
- moosh version: 2015092200
- moodle version: 2.8 (upgrading to 3.1 by September)
- php version: 5.5.9 (upgrading to 7 by September)
- operating system: Ubuntu 14.04

We are currently revising our Moodle processes and using Moosh for a number of repetitive tasks.

Among other tasks, we wish to use the course-backup and course-restore commands. The current version off CourseRestore.php has no feature to overwrite an existing course (instead of adding to the current content of the destination course).

I am aware that a new CourseRestore.php has been created by @agru (Pull #118 ) to include a new option -o, which deletes the course content of the existing course before adding the content of the backup file.

We'd really see value if this feature is included in the next Moosh release.

Meanwhile, is there a way that we cantry out this updated command ourselves on Moodle 2.8, and possibly later try it out on Moodle 3.1 when that is released? I would really appreciate some guidance.

(apologies for not being too technical, I'm not too much of a tech person)

Thanks,
Chris

**Comment by dkns** [comment]
So if I understand correctly you want to use feature that @agru added to CourseRestore.php but you're on version that doesn't have that functionality yet?

In that case you can install moosh via git, instructions here http://moosh-online.com/ This will give you the latest version of moosh with all the commands, features, etc.

**Comment by csch1** [comment]
Thanks for the reply! 

We managed to sort it out by copying Moodle29/Course/ to Moodle28/, and updating the namespace in each of the four scripts.

In CourseRestore.php, the folllowing entry in line 167 was updated by setting it to ‘1’ in order to retain enrolled users in the destination area:
            $deletingoptions['keep_roles_and_enrolments'] = 1;

Preliminary tests indicate that the script is functioning. We will report back with any findings following more testing.

Thinks,
Chris
